examples/DotNetCore/ConsoleAppWithTableStorageReference/Program.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Data.Tables;
+using Azure.Identity;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Examples.ConsoleAppWithTableStorageReference
+{
+    // C# record type for items in the table
+    public record Product : ITableEntity
+    {
+        public string RowKey { get; set; } = default!;
+
+        public string PartitionKey { get; set; } = default!;
+
+        public string Name { get; init; } = default!;
+
+        public int Quantity { get; init; }
+
+        public bool Sale { get; init; }
+
+        public ETag ETag { get; set; } = default!;
+
+        public DateTimeOffset? Timestamp { get; set; } = default!;
+    }
+
+    class Program
+    {
+        static IConfiguration Configuration { get; set; }
+
+        static void Main(string[] args)
+        {
+            Configure();
+
+            string display = string.Empty;
+            StringBuilder sb = new StringBuilder();
+
+            List<Product> tableContent = new List<Product>();
+            Configuration.GetSection("TableUri").Bind(tableContent);
+
+            sb.AppendLine("Your store inventory:\n");
+
+            var tableFormat = "{0,-20} {1,-10} {2,-10}";
+
+            sb.AppendFormat(tableFormat, "Product Name", "Quantity", "On Sale");
+            sb.AppendLine();
+
+            foreach (Product product in tableContent)
```

Comment:
Nothing wrong with this, but it kind of counters the purpose of showing off the configuration. I would just show the whole inventory based on the configuration (everything under the "Inventory" node).

Reply:
Not sure I understand, are you saying instead of how I bound it to `tableContent`, just to print the configuration section to the console?

Reply:
Right. While the data is saved as a table in storage, the whole idea is to make it available as a configuration. Your application should be dealing with configuration, not a table. I can provide that same data in `appsettings.json` like below and there should be no difference to the core business of an application, which is displaying the data in your example.

```json
"MyShop" : {
    "Inventory" : [
        {
            "ProductName": "Foo",
            "Quantity": 98,
            "OnSale" : true
        },
        {
            "ProductName": "bar",
            "Quantity": 1,
            "OnSale" : false
        }
    ]
}
```

Reply:
From what I've tested, even with binding the data to a list of products this kind of setup from `appsettings.json` should yield the same result as long as the keys are the same. From what I can tell the inventory value is a list so if I just print the value it's null unless I bind it this way, so let me know if this works. I've added a similar example to the one you commented to `appsettings.json` as well.

Reply:
@zhenlan I think we should still bind configuration to `List<Product>` because I assume that's how real applications would use it.

But this example can be simplified further if we dont display the config as a table. How about something like this:

```cs
foreach (Product product in tableContent)
{
    if (columnsToDisplay.Contains(nameof(product.Name)))
    {
        sb.AppendLine($"Name = {product.Name}");
    }
   
    if (columnsToDisplay.Contains(nameof(product.Quantity)))
    {
        sb.AppendLine($"Quantity = {product.Quantity.ToString()}");
    }

    if (columnsToDisplay.Contains(nameof(product.OnSale)))
    {
        sb.AppendLine($"OnSale = {product.OnSale.ToString()}");
    }

    sb.AppendLine();
}

```